### PR TITLE
#1864 - PT Integration Files Review - MSFAA Send/Request Files

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- MSFAA sent/received schedulers were named incorrectly (the request was a response and vice-versa). There are no actual changes on files `msfaa-integration/msfaa-part-time-process-integration.scheduler.ts` and `msfaa-part-time-process-response-integration.scheduler.ts` other than the file renaming.
- Replaced some deprecated `substr` by `substring`.
- Added a condition `cancelledDate is null` to pending MSFAA generation to avoid generating an MSFAA to canceled records.

### CODE REVIEW

File: sources/packages/backend/libs/integrations/src/esdc-integration/msfaa-integration/msfaa-files/msfaa-file-header.ts [hunk 1]
```diff
@@ -2,11 +2,12 @@ import {
   DATE_FORMAT,
   MSFAARequestFileLine,
   MSFAA_SENT_TITLE,
+  NUMBER_FILLER,
   RecordTypeCodes,
   SPACE_FILLER,
+  TIME_FORMAT,
```
Comment (guru-aot) [inline sources/packages/backend/libs/integrations/src/esdc-integration/msfaa-integration/msfaa-files/msfaa-file-header.ts:8]: Nice one ❤️ .

File: sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts [hunk 2]
```diff
@@ -54,6 +54,7 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
       .innerJoin("currentAssessment.offering", "offering")
       .innerJoin("offering.institutionLocation", "institutionLocation")
       .where("msfaaNumber.dateRequested is null")
+      .andWhere("msfaaNumber.cancelledDate is null")
```
Comment (guru-aot) [inline sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts:57]: Curious, the generation of pending msfaa for a cancelled record, should have a dateRequested already right? or are we considering the old data that was imported for MSFAA?

**Review by guru-aot — APPROVED** [review]
LGTM, thanks for doing the refactoring and replacements.

**Review by andrepestana-aot — APPROVED** [review]
LGTM.

File: sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts:57]: Mine though was to add this condition in preparation for the upcoming requirement for MSFAA reactivation which would potentially have SIMS setting the canceled date.
From #1777 
_Update ANY OTHER active (with null cancellation) MSFAA numbers for the student for the SAME OFFERING INTENSITY by populating the cancelled date and a boolean column indicating it is a system cancellation._

Either way, sending a canceled MSFAA would be a mistake.

**Review by ann-aot — APPROVED** [review]
👍

File: sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts (hunk 2 above)
Comment (dheepak-aot) [reply sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts:57]: @guru-aot with the upcoming requirement of re-activating MSFAA number, A cancelled MSFAA number can be re-activated where a pending msfaa(date_signed is null) could be cancelled by SIMS. At this situation, for that msfaa record which is cancelled will never have date signed.

**Review by dheepak-aot — APPROVED** [review]
Thanks for taking up the substring refactors and also getting the schedulers right. LGTM. 👍

**Pull request merged**